Extension/src/LanguageServer/configurations.ts

```diff
@@ -598,9 +601,22 @@ export class CppProperties {
         try {
             filePaths.forEach((path: string) => {
                 this.compileCommandFileWatchers.push(fs.watch(path, (event: string, filename: string) => {
-                    if (event !== "rename") {
-                        this.onCompileCommandsChanged(path);
+                    if (event === "rename") {
+                        return;
+                    }
+                    // Wait 1 second after a change to allow time for the write to finish.
+                    if (this.compileCommandsFileWatcherTimer) {
+                        clearInterval(this.compileCommandsFileWatcherTimer);
                     }
+                    this.compileCommandsFileWatcherFiles.add(path);
```

Comment:
looks like compileCommandsFileWatcherFiles may get multiple copies of "path" if file change events are trigger within quick successions. Maybe we need to check if "path" already exists or simply clear the list in the same clock as clearInterval above ?

Reply:
o never mind, i didn't notice it was a Set<string> :)